synapse/federation/transaction_queue.py

```diff
@@ -211,13 +229,19 @@ def _send_new_transaction(self, destination, pending_pdus, pending_edus,
                     self.store,
                 )
 
+                device_message_edus, device_stream_id = (
+                    yield self._get_new_device_messages(destination)
+                )
```

Comment:
It's very strange to be pulling out here, given we check in `_attempt_new_transaction` whether there is anything to send.